app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Admin::ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+  end
+
+  def update
+    application = Application.find(params[:id])
+    pet_app = PetApplication.find_by(pet_id: params[:pet_id])
```

Comment:
You can send more than one param to find_by, so you could do `PetApplication.find_by(pet_id: params[:pet_id], application_id: params[:application_id])`